### PR TITLE
Error when calling methods on MetaType

### PR DESCRIPTION
**Pull request by jez**
### Test plan

See included automated tests.

### CODE REVIEW

**Comment by jez** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build result here:

→ https://go/builds/bui_I1tT9rOwh7aD7r

File: core/types/calls.cc [hunk 1]
```diff
@@ -1002,6 +1002,18 @@ DispatchResult MetaType::dispatchCall(const GlobalState &gs, DispatchArgs args)
...
+                            this->wrapped->show(gs));
+                if (args.name == core::Names::tripleEq()) {
+                    if (auto appliedType = cast_type<AppliedType>(this->wrapped.get())) {
+                        e.addErrorSection(
+                            ErrorSection("It looks like you're trying to pattern match on a generic", {}));
+                        e.replaceWith("Replace with class name", loc, "{}", appliedType->klass.data(gs)->show(gs));
+                    }
+                }
+            }
             return ProxyType::dispatchCall(gs, args);
```
Comment (jez) [inline core/types/calls.cc:1017]: oh whoops, meant to delete this line and return untyped; i don't think it ever makes sense to continue to dispatch the call onto the underlying value.

File: test/cli/autocorrect-case-meta-type/autocorrect-case-meta-type.out [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  when Integer
+    input
+  when Array
+    input.first || 0
+  else
+    T.absurd(input)
+  end
+end
+
+Array
```
Comment (jez) [inline test/cli/autocorrect-case-meta-type/autocorrect-case-meta-type.out:48]: You can see the raw `.===` call gets kind of clobbered, but I'm ok with that.

**Review by elliottt — APPROVED** [review]
This is great!

**Pull request merged**